### ISSUE TITLE
Crashes not showing in fabric portal

### ISSUE THREAD

**Issue by jgtaveras** [drungrin/Fabric.Sdk.Xamarin#44]
First of all great work with the bindings

For some reason crashes are not being shown in the fabric's portal, the console output says it uploaded the report successfully but nothing is actually shown

Here is the console output:

`[CrashlyticsCore] Adding single file 59710C8B0282-0001-358F-6A97E83BAD68.cls to report 59710C8B0282-0001-358F-6A97E83BAD68
[CrashlyticsCore] Sending report to: https://reports.crashlytics.com/spi/v1/platforms/android/apps/com.mysample.app/reports
[CrashlyticsCore] Create report request ID: null
[CrashlyticsCore] Result was: 202
[CrashlyticsCore] Crashlytics report upload complete: 59710C8B0282-0001-358F-6A97E83BAD68
[CrashlyticsCore] Removing report at /data/data/com.mysample.app/files/.Fabric/com.crashlytics.sdk.android.crashlytics-core/fatal-sessions/59710C8B0282-0001-358F-6A97E83BAD68.cls`

**Comment by OmidID** [comment]
Hello @jgtaveras,
Did you solved the problem?

**Comment by HenrikPihl** [comment]
I have the same problem. Could it be related to that the Crashlytics SDK have been updated?

**Comment by sergiovenancio** [comment]
Same here... I've tried Crashlytics.Instance.Crash() to force a crash, but no hits. Answers is working well, though.

**Comment by Annelies1** [comment]
I have the same issue. Does anyone have a solution yet?

This is the log output:

[Fabric] Build ID is: 973e36cc-9e22-44c5-8ca5-bb430f6e13fb
[CrashlyticsCore] Initializing Crashlytics 2.3.16.dev
[Beta] Beta by Crashlytics app is not installed
[Beta] Beta device token load took 16.538ms
[Beta] Beta device token present: false
[CrashlyticsCore] Installer package name is: null
[CrashlyticsCore] Found previous crash marker.
[CrashlyticsCore] Exception handling initialization successful
[Fabric] Initializing io.fabric.sdk.android:fabric [Version: 1.3.16.dev], with the following kits:
[Fabric] com.crashlytics.sdk.android:beta [Version: 1.2.4.dev]
[Fabric] com.crashlytics.sdk.android:answers [Version: 1.3.12.dev]
[Fabric] com.crashlytics.sdk.android:crashlytics [Version: 2.6.7.dev]
[Fabric] com.crashlytics.sdk.android.crashlytics-core [Version: 2.3.16.dev]

......

[CrashlyticsCore] Checking for crash reports...
[CrashlyticsCore] Found crash report /data/user/0/myapp.bundle.id/files/.Fabric/com.crashlytics.sdk.android.crashlytics-core/fatal-sessions/5A3BC49E036E-0001-0C34-22086C1EF86F.cls
[CrashlyticsCore] Attempting to send 1 report(s)
[CrashlyticsCore] Adding single file 5A3BC49E036E-0001-0C34-22086C1EF86F.cls to report 5A3BC49E036E-0001-0C34-22086C1EF86F
[CrashlyticsCore] Sending report to: https://reports.crashlytics.com/spi/v1/platforms/android/apps/myapp.bundle.id/reports
[CrashlyticsCore] Create report request ID: null
[CrashlyticsCore] Result was: 202
[CrashlyticsCore] Crashlytics report upload complete: 5A3BC49E036E-0001-0C34-22086C1EF86F
[CrashlyticsCore] Removing report at /data/user/0/myapp.bundle.id/files/.Fabric/com.crashlytics.sdk.android.crashlytics-core/fatal-sessions/5A3BC49E036E-0001-0C34-22086C1EF86F.cls
[CrashlyticsCore] Checking for crash reports...
[CrashlyticsCore] No reports found.


Edit: I got it working now. it was the bundle id that didn't match with the android studio project i made before. also i made sure now that the android studio project created a crash, before creating a crash in the xamarin project.

**Comment by guilhemvors** [comment]
I have the same problem:

`03-08 16:31:24.318 11124-11165/? D/Beta: Beta by Crashlytics app is not installed
03-08 16:31:24.323 11124-11124/? D/CrashlyticsNdk: Crashlytics NDK initialization successful
03-08 16:31:24.325 11124-11124/? I/CrashlyticsCore: Initializing Crashlytics 2.6.1.23
03-08 16:31:24.328 11124-11124/? D/CrashlyticsCore: Installer package name is: null
03-08 16:31:24.339 11124-11124/? D/CrashlyticsCore: Exception handling initialization successful
03-08 16:31:24.340 11124-11169/? D/CrashlyticsCore: Opening a new session with ID 5AA1574C0152-0001-2B74-8BC6899AFCB9
03-08 16:31:24.724 11124-11169/? D/CrashlyticsCore: Initialization marker file created.
03-08 16:31:25.038 11124-11169/? D/CrashlyticsCore: Could not generate ID for file /dev/ashmem/dalvik-jit-code-cache (deleted)
                                                    java.io.FileNotFoundException: /dev/ashmem/dalvik-jit-code-cache (deleted) (Not a directory)
                                                        at java.io.FileInputStream.open0(Native Method)
                                                        at java.io.FileInputStream.open(FileInputStream.java:200)
                                                        at java.io.FileInputStream.<init>(FileInputStream.java:150)
                                                        at java.io.FileInputStream.<init>(FileInputStream.java:103)
                                                        at com.crashlytics.android.core.Sha1FileIdStrategy.getFileSHA(Sha1FileIdStrategy.java:28)
                                                        at com.crashlytics.android.core.Sha1FileIdStrategy.createId(Sha1FileIdStrategy.java:17)
                                                        at com.crashlytics.android.core.BinaryImagesConverter.parseProcMapsJson(BinaryImagesConverter.java:67)
                                                        at com.crashlytics.android.core.BinaryImagesConverter.convert(BinaryImagesConverter.java:33)
                                                        at com.crashlytics.android.core.NativeFileUtils.processBinaryImages(NativeFileUtils.java:85)
                                                        at com.crashlytics.android.core.NativeFileUtils.binaryImagesJsonFromFile(NativeFileUtils.java:72)
                                                        at com.crashlytics.android.core.NativeFileUtils.binaryImagesJsonFromDirectory(NativeFileUtils.java:62)
                                                        at com.crashlytics.android.core.CrashlyticsController.finalizeMostRecentNativeCrash(CrashlyticsController.java:901)
                                                        at com.crashlytics.android.core.CrashlyticsController.access$1700(CrashlyticsController.java:68)
                                                        at com.crashlytics.android.core.CrashlyticsController$16.call(CrashlyticsController.java:971)
                                                        at com.crashlytics.android.core.CrashlyticsController$16.call(CrashlyticsController.java:961)
                                                        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
                                                        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
                                                        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
                                                        at io.fabric.sdk.android.services.common.ExecutorUtils$1$1.onRun(ExecutorUtils.java:75)
                                                        at io.fabric.sdk.android.services.common.BackgroundPriorityRunnable.run(BackgroundPriorityRunnable.java:30)
                                                        at java.lang.Thread.run(Thread.java:764)
03-08 16:31:25.617 11124-11169/? D/CrashlyticsCore: Finalizing previously open sessions.
03-08 16:31:25.619 11124-11169/? D/CrashlyticsCore: Closing open sessions.
03-08 16:31:25.619 11124-11169/? D/CrashlyticsCore: Closing session: 5AA15640007A-0001-2AEF-8BC6899AFCB9
03-08 16:31:25.619 11124-11169/? D/CrashlyticsCore: Collecting session parts for ID 5AA15640007A-0001-2AEF-8BC6899AFCB9
03-08 16:31:25.619 11124-11169/? D/CrashlyticsCore: Session 5AA15640007A-0001-2AEF-8BC6899AFCB9 has fatal exception: false
03-08 16:31:25.620 11124-11169/? D/CrashlyticsCore: Session 5AA15640007A-0001-2AEF-8BC6899AFCB9 has non-fatal exceptions: false
03-08 16:31:25.620 11124-11169/? D/CrashlyticsCore: No events present for session ID 5AA15640007A-0001-2AEF-8BC6899AFCB9
03-08 16:31:25.620 11124-11169/? D/CrashlyticsCore: Removing session part files for ID 5AA15640007A-0001-2AEF-8BC6899AFCB9
03-08 16:31:25.621 11124-11169/? D/CrashlyticsCore: Closed all previously open sessions
03-08 16:31:25.622 11124-11190/? D/CrashlyticsCore: Starting report processing in 1.0 second(s)...
03-08 16:31:25.623 11124-11169/? D/CrashlyticsCore: Initialization marker file removed: true
03-08 16:31:26.634 11124-11190/? D/CrashlyticsCore: Checking for crash reports...
03-08 16:31:26.640 11124-11190/? D/CrashlyticsCore: Attempting to send 1 report(s)
03-08 16:31:27.002 11124-11190/? D/CrashlyticsCore: Sending report to: https://reports.crashlytics.com/sdk-api/v1/platforms/android/apps/com.myCompany.myAppandroid/minidumps
03-08 16:31:27.299 11124-11190/? D/CrashlyticsCore: Result was: 202
03-08 16:31:27.300 11124-11190/? I/CrashlyticsCore: Crashlytics report upload complete: 5AA15640007A-0001-2AEF-8BC6899AFCB9
03-08 16:31:27.300 11124-11190/? D/CrashlyticsCore: Removing native report file at /data/user/0/com.myCompany.myAppandroid/files/.Fabric/com.crashlytics.sdk.android.crashlytics-core/5AA15640007A-0001-2AEF-8BC6899AFCB9/minidump
03-08 16:31:27.301 11124-11190/? D/CrashlyticsCore: Removing native report file at /data/user/0/com.myCompany.myAppandroid/files/.Fabric/com.crashlytics.sdk.android.crashlytics-core/5AA15640007A-0001-2AEF-8BC6899AFCB9/metadata
03-08 16:31:27.302 11124-11190/? D/CrashlyticsCore: Removing native report file at /data/user/0/com.myCompany.myAppandroid/files/.Fabric/com.crashlytics.sdk.android.crashlytics-core/5AA15640007A-0001-2AEF-8BC6899AFCB9/binaryImages
03-08 16:31:27.303 11124-11190/? D/CrashlyticsCore: Removing native report file at /data/user/0/com.myCompany.myAppandroid/files/.Fabric/com.crashlytics.sdk.android.crashlytics-core/5AA15640007A-0001-2AEF-8BC6899AFCB9/session
03-08 16:31:27.303 11124-11190/? D/CrashlyticsCore: Removing native report file at /data/user/0/com.myCompany.myAppandroid/files/.Fabric/com.crashlytics.sdk.android.crashlytics-core/5AA15640007A-0001-2AEF-8BC6899AFCB9/app
03-08 16:31:27.304 11124-11190/? D/CrashlyticsCore: Removing native report file at /data/user/0/com.myCompany.myAppandroid/files/.Fabric/com.crashlytics.sdk.android.crashlytics-core/5AA15640007A-0001-2AEF-8BC6899AFCB9/device
03-08 16:31:27.305 11124-11190/? D/CrashlyticsCore: Removing native report file at /data/user/0/com.myCompany.myAppandroid/files/.Fabric/com.crashlytics.sdk.android.crashlytics-core/5AA15640007A-0001-2AEF-8BC6899AFCB9/os
03-08 16:31:27.305 11124-11190/? D/CrashlyticsCore: Removing native report directory at /data/user/0/com.myCompany.myAppandroid/files/.Fabric/com.crashlytics.sdk.android.crashlytics-core/5AA15640007A-0001-2AEF-8BC6899AFCB9
03-08 16:31:27.306 11124-11190/? D/CrashlyticsCore: Checking for crash reports...
03-08 16:31:27.311 11124-11190/? D/CrashlyticsCore: No reports found.`

Some additional information:

- I can see java crashes, but not C++ one.
- at some point I had a different package name for my app, and I could see the NDK crashes as well. I changed the package name but kept the APIKey, as it is company wide.
- when building in debug, I don't see the crash being sent at, all, even if the java crashes are reported

**Comment by ashu-mkb** [comment]
@Annelies1 which bundle id? i am running into the same problem, help please.

**Comment by Annelies1** [comment]
@ashu-mkb the package name e.g. com.company.appname . use android studio first to create a project with your package name and create a crash there and check if it is logged. after that you can use it in xamarin.

**Comment by ashu-mkb** [comment]
Actually i am using android studio (not a xamarin dev). This is so awkward as its been a week and still i am not able to figure it out. I have checked package name and it matches with the crashlytics one. I also tried contacting support, reply is still awaited as of now.

**Comment by yterletskyi** [comment]
did you manage to fix the problem? I'm facing the it too

**Comment by ashu-mkb** [comment]
@yterletskyi Somehow project json file was outdated. After enabling crashlytics i need to download updated json from firebase and sync. That worked for me.

**Comment by yterletskyi** [comment]
Thanks @ashu-mkb, but for some reason it's still not working. I see `Crashlytics report upload complete: <ID>` in my logcat but when I sniffed the network it appeared that request isn't successful -- it's 403 Forbiden and I receive an exception in json. Is anyone here from Fabric or Firebase support?

**Comment by tarunmehta-quovantis** [comment]
did anyone find the solution?

**Comment by jayminnayak** [comment]
I am having the same issue. I have created projects for Android and iOS using android studio and Xcode. Both the applications are registered successfully on the Fabric Portal but it does not log the crash when I run the application in Xamarin.

Does anybody find a solution?

**Comment by ssuchanowski** [comment]
We found out that for our case (moving from native iOS to [react-native-firebase](https://github.com/invertase/react-native-firebase) lib) if app was previously in Fabric and migrated then we need to leave the original configuration in (being here: API key in info.plist and api key + secret in run script config).

I dont think this is the RN library fault - it might be that Firebase doesn't migrate the crashlytics keys somehow. Not sure. We've spent way too much time on this so I hope this helps.
